src/Microsoft.ML.Core/Utilities/LineParser.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.ML.Runtime.Internal.Utilities
+{
+    [BestFriend]
+    internal static class LineParser
+    {
+        [BestFriend] // required because the return type is a value tuple..
```

Comment:
I'm not sure I understand this comment. Shouldn't the BestFriend attribute on the class be enough?

Maybe keep the class `internal`, but make the method `public`?

/cc @TomFinley

Reply:
It seems that it treats a ValueTuple as defined in this library and adding the `[BestFriend]` for the method makes it work. If I remove it, it does not work.

Reply:
Wow, that's super odd and unexpected IMO.

Even if you make the method `public` it doesn't work?

Reply:
hah, with public it works ;)